eras/conway/impl/src/Cardano/Ledger/Conway/TxInfo.hs

```diff
@@ -10,183 +11,301 @@
-      , PV3.txInfoReferenceInputs = refInputs
-      , PV3.txInfoFee = Alonzo.transValue (inject @_ @(MaryValue (EraCrypto era)) fee)
-      , -- Note that this translation is different from previous Plutus versions, since we no
-        -- longer add a zero ADA value to the mint field during translation:
-        PV3.txInfoMint = Alonzo.transMultiAsset (txBody ^. mintTxBodyL)
-      , PV3.txInfoTxCerts = toList $ fmap (unTxCertV3 . Alonzo.transTxCert) (txBody ^. certsTxBodyL)
-      , PV3.txInfoWdrl = PV3.fromList $ Map.toList (transWithdrawals (txBody ^. withdrawalsTxBodyL))
-      , PV3.txInfoValidRange = timeRange
-      , PV3.txInfoSignatories =
-          map Alonzo.transKeyHash (Set.toList (txBody ^. reqSignerHashesTxBodyL))
-      , PV3.txInfoRedeemers = PV3.fromList rdmrs'
-      , PV3.txInfoData = PV3.fromList $ map Alonzo.transDataPair datpairs
-      , PV3.txInfoId = PV3.TxId (Alonzo.transSafeHash (hashAnnotated txBody))
-      , -- FIXME: implement for plutus v3
-        PV3.txInfoVotes = error "Unimplemented"
-      , -- FIXME: implement for plutus v3
-        PV3.txInfoProposalProcedures = error "Unimplemented"
-      , -- FIXME: implement for plutus v3
-        PV3.txInfoCurrentTreasuryAmount = error "Unimplemented"
-      , -- FIXME: implement for plutus v3
-        PV3.txInfoTreasuryDonation = error "Unimplemented"
-      }
-  where
-    txBody = tx ^. bodyTxL
-    witnesses = tx ^. witsTxL
-    outs = txBody ^. outputsTxBodyL
-    fee = txBody ^. feeTxBodyL
-    datpairs = Map.toList (unTxDats $ witnesses ^. datsTxWitsL)
-    rdmrs = Map.toList (unRedeemers $ witnesses ^. rdmrsTxWitsL)
-
--- | This is a temporary version that only translates certificates from previous eras,
--- none of them are Conway specific.
-conwayTransTxCert :: ShelleyEraTxCert era => TxCert era -> PV3.TxCert
-conwayTransTxCert = \case
+instance Crypto c => EraPlutusContext (ConwayEra c) where
+  data ContextError (ConwayEra c)
+    = BabbageContextError !(ContextError (BabbageEra c))
+    | CertificateNotSupported !(TxCert (ConwayEra c))
+    deriving (Eq, Show, Generic)
+
+  mkPlutusScriptContext = \case
+    ConwayPlutusV1 p -> mkPlutusLanguageContext p
+    ConwayPlutusV2 p -> mkPlutusLanguageContext p
+    ConwayPlutusV3 p -> mkPlutusLanguageContext p
+
+instance Inject (ContextError (ConwayEra c)) (ContextError (ConwayEra c))
+
+instance Inject (ContextError (BabbageEra c)) (ContextError (ConwayEra c)) where
+  inject = BabbageContextError
+
+instance Inject (ContextError (AlonzoEra c)) (ContextError (ConwayEra c)) where
+  inject = BabbageContextError . inject
+
+instance NoThunks (ContextError (ConwayEra c))
+
+instance Crypto c => NFData (ContextError (ConwayEra c))
+
+instance Crypto c => EncCBOR (ContextError (ConwayEra c)) where
+  encCBOR = \case
+    -- We start at tag 8, just in case to avoid clashes with previous eras.
+    BabbageContextError babbageContextError ->
+      encode $ Sum BabbageContextError 8 !> To babbageContextError
```

Comment:
Why are we encoding an additional tag here? Couldn't we just use the same tag as the final error that we're actually encoding?

Reply:
I don't understand your question. Could you make a code suggestion, so it is clear what you mean?